implementation/leprechaun/src/shared/api/generic-paginated.dto.ts

```diff
@@ -0,0 +1,30 @@
+import { ApiProperty } from '@nestjs/swagger';
+
+export class PaginatedDto<TDto> {
+  readonly data: TDto[];
+
+  @ApiProperty({ description: 'Total number of items', example: 420, minimum: 0 })
+  readonly total: number;
+  @ApiProperty({ description: 'Number of items per page', example: 30, minimum: 1, maximum: 100, default: 10 })
+  readonly size: number;
+  @ApiProperty({ description: 'Current page number', example: 12, minimum: 0, default: 0 })
+  readonly page: number;
+
+  constructor(total, size, page, data) {
+    this.total = total;
+    this.size = size;
+    this.page = page;
+    this.data = data;
+  }
+
+  static fromEntities<TDto>(total: number, size: number, page: number, entities: TDto[]): PaginatedDto<TDto> {
```

Comment:
**very very nit:** You are calling this method `fromEntities` and it takes DTOs instead of entities as arguments.

Reply:
Yeah, naming problem.